### ISSUE TITLE
CoRC system compatibility is scarcely tested

### ISSUE THREAD

**Issue by jonasfoe** [jonasfoe/copasi-r-api#4]
We are currently unsure whether CoRC is stable on all system configurations.
Currently no system is known to cause crashes.
32 bit systems are unsupported.
It would be very helpful to get some feedback on which systems can fully execute the test set / where it fails.

Testing CoRC can be done as follows:
```
install.packages(c("remotes", "sessioninfo", "testthat"))
remotes::install_github("jpahle/CoRC", force = TRUE, INSTALL_opts = "--install-tests")
CoRC::getCopasi()
sessioninfo::session_info("CoRC")
testthat::test_package("CoRC")
```

The output the last two commands is of interest.

**Comment by jonasfoe** [comment]
```
> devtools::session_info()
Session info -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.2 (2017-09-28)
 system   x86_64, mingw32             
 ui       RStudio (1.1.383)           
 language (EN)                        
 collate  English_United States.1252  
 tz       Europe/Berlin               
 date     2017-11-25                  

Packages -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 package    * version    date       source                      
 assertthat   0.2.0      2017-04-11 CRAN (R 3.4.0)              
 backports    1.1.1      2017-09-25 CRAN (R 3.4.1)              
 base       * 3.4.2      2017-09-28 local                       
 compiler     3.4.2      2017-09-28 local                       
 CoRC         0.0.1.9001 2017-11-25 Github (jpahle/CoRC@1ba058e)
 curl         3.0        2017-10-06 CRAN (R 3.4.2)              
 datasets   * 3.4.2      2017-09-28 local                       
 devtools     1.13.4     2017-11-09 CRAN (R 3.4.2)              
 digest       0.6.12     2017-01-27 CRAN (R 3.4.0)              
 git2r        0.19.0     2017-07-19 CRAN (R 3.4.1)              
 graphics   * 3.4.2      2017-09-28 local                       
 grDevices  * 3.4.2      2017-09-28 local                       
 httr         1.3.1      2017-08-20 CRAN (R 3.4.1)              
 isnullptr    1.0.1      2017-09-28 CRAN (R 3.4.2)              
 knitr        1.17       2017-08-10 CRAN (R 3.4.1)              
 magrittr     1.5        2014-11-22 CRAN (R 3.4.0)              
 memoise      1.1.0      2017-04-21 CRAN (R 3.4.0)              
 methods    * 3.4.2      2017-09-28 local                       
 purrr        0.2.4      2017-10-18 CRAN (R 3.4.2)              
 R6           2.2.2      2017-06-17 CRAN (R 3.4.0)              
 rlang        0.1.4      2017-11-05 CRAN (R 3.4.2)              
 stats      * 3.4.2      2017-09-28 local                       
 stringi      1.1.6      2017-11-17 CRAN (R 3.4.2)              
 stringr      1.2.0      2017-02-18 CRAN (R 3.4.0)              
 tools        3.4.2      2017-09-28 local                       
 utils      * 3.4.2      2017-09-28 local                       
 withr        2.1.0      2017-11-01 CRAN (R 3.4.2)              
 yaml         2.1.14     2016-11-12 CRAN (R 3.4.0)              
> testthat::test_package("CoRC")
Entity accessors: ..................................
Entity finders: ..............................................................................................
Expressions: .......................
Task Linear Noise Approximation: .........
Task Metabolic Control Analysis: .....................
Task Optimization: ............
Task Parameter Estimation: ...................................
Task SteadyState: ................
Task TimeCourse: ...........

DONE ======================================================================================================================================================================================================================

```

**Comment by jonasfoe** [comment]
```
> devtools::session_info()
Session info ------------------------------------------------------------------
 setting  value
 version  R version 3.3.1 (2016-06-21)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.utf8
 tz       <NA>
 date     2017-11-25

Packages ----------------------------------------------------------------------
 package    * version    date       source
 assertthat   0.2.0      2017-04-11 cran (@0.2.0)
 backports    1.1.1      2017-09-25 cran (@1.1.1)
 base       * 3.3.1      2016-08-08 local
 CoRC         0.0.1.9001 2017-11-25 Github (jpahle/CoRC@1ba058e)
 curl         3.0        2017-10-06 CRAN (R 3.3.1)
 datasets   * 3.3.1      2016-08-08 local
 devtools     1.13.4     2017-11-09 CRAN (R 3.3.1)
 digest       0.6.12     2017-01-27 CRAN (R 3.3.1)
 git2r        0.19.0     2017-07-19 CRAN (R 3.3.1)
 graphics   * 3.3.1      2016-08-08 local
 grDevices  * 3.3.1      2016-08-08 local
 httr         1.3.1      2017-08-20 CRAN (R 3.3.1)
 isnullptr    1.0.1      2017-09-28 cran (@1.0.1)
 magrittr     1.5        2014-11-22 cran (@1.5)
 memoise      1.1.0      2017-04-21 CRAN (R 3.3.1)
 methods    * 3.3.1      2016-08-08 local
 purrr        0.2.4      2017-10-18 cran (@0.2.4)
 R6           2.2.2      2017-06-17 CRAN (R 3.3.1)
 rlang        0.1.4      2017-11-05 cran (@0.1.4)
 stats      * 3.3.1      2016-08-08 local
 stringi      1.1.5      2017-04-07 cran (@1.1.5)
 stringr      1.2.0      2017-02-18 cran (@1.2.0)
 tools        3.3.1      2016-08-08 local
 utils      * 3.3.1      2016-08-08 local
 withr        2.1.0      2017-11-01 CRAN (R 3.3.1)
> testthat::test_package("CoRC")
Entity accessors: ..................................
Entity finders: ..............................................................................................
Expressions: .......................
Task Linear Noise Approximation: .........
Task Metabolic Control Analysis: .....................
Task Optimization: ............
Task Parameter Estimation: ...................................
Task SteadyState: ................
Task TimeCourse: ...........W

Warnings -----------------------------------------------------------------------
1. autoplot.copasi_ts() (@test_task_timecourse.R#32) - Skipping autoplot test because package 'ggplot2' is missing

DONE ===========================================================================

```

**Comment by jonasfoe** [comment]
```
> devtools::session_info()
Session info --------------------------------------------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.3.3 (2017-03-06)
 system   x86_64, darwin13.4.0        
 ui       RStudio (1.0.136)           
 language (EN)                        
 collate  de_DE.UTF-8                 
 tz       Europe/Berlin               
 date     2017-11-27                  

Packages ------------------------------------------------------------------------------------------------------------------------------------------
 package    * version    date       source                      
 assertthat   0.2.0      2017-04-11 cran (@0.2.0)               
 backports    1.1.1      2017-09-25 cran (@1.1.1)               
 base       * 3.3.3      2017-03-07 local                       
 CoRC         0.0.1.9001 2017-11-27 Github (jpahle/CoRC@1ba058e)
 curl         3.0        2017-10-06 CRAN (R 3.3.2)              
 datasets   * 3.3.3      2017-03-07 local                       
 devtools     1.13.4     2017-11-09 CRAN (R 3.3.2)              
 digest       0.6.12     2017-01-27 CRAN (R 3.3.2)              
 git2r        0.19.0     2017-07-19 CRAN (R 3.3.2)              
 graphics   * 3.3.3      2017-03-07 local                       
 grDevices  * 3.3.3      2017-03-07 local                       
 httr         1.3.1      2017-08-20 CRAN (R 3.3.2)              
 isnullptr    1.0.1      2017-09-28 cran (@1.0.1)               
 magrittr     1.5        2014-11-22 cran (@1.5)                 
 memoise      1.1.0      2017-04-21 CRAN (R 3.3.2)              
 methods    * 3.3.3      2017-03-07 local                       
 purrr        0.2.4      2017-10-18 cran (@0.2.4)               
 R6           2.2.2      2017-06-17 CRAN (R 3.3.2)              
 rlang        0.1.4      2017-11-05 cran (@0.1.4)               
 stats      * 3.3.3      2017-03-07 local                       
 stringi      1.1.5      2017-04-07 cran (@1.1.5)               
 stringr      1.2.0      2017-02-18 cran (@1.2.0)               
 tools        3.3.3      2017-03-07 local                       
 utils      * 3.3.3      2017-03-07 local                       
 withr        2.1.0      2017-11-01 CRAN (R 3.3.2)              
> testthat::test_package("CoRC")
Entity accessors: ..................................
Entity finders: ..............................................................................................
Expressions: .......................
Task Linear Noise Approximation: .........
Task Metabolic Control Analysis: .....................
Task Optimization: ............
Task Parameter Estimation: ...................................
Task SteadyState: ................
Task TimeCourse: ...........

DONE ===============================================================================================================================================
```

**Comment by radivot** [comment]
Wondering if you could guide me to getting this code to give time course results.
The cps files are here [cpsFiles.zip](https://github.com/jpahle/CoRC/files/2901333/cpsFiles.zip)


library(CoRC)
unloadAllModels()
(L=loadExamples())
runTC(model=L[[1]])  #runs fine
path="~/ccf/jarek/grants/msb/iron/parmar19sup/cps/"
(m0=loadModel(paste0(path,"IronMousePV3.cps")))
runTC(model=m0) # no results
(m1=loadModel(paste0(path,"IronMousePV3_Hemochromatosis.cps")))
runTC(model=m1) # no results

**Comment by jonasfoe** [comment]
I have tested the package with windows, macos, ubuntu, centos, and fedora.
Previous issues with centos have been resolved (807b7be9301a951fd1d281253350fee0203fe6df).
